### PR TITLE
feat: add take-from-basket command

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Ref: #732 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/client/basket/tx.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+			if err := msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), &msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+	cmd.Flags().String(FlagRetirementLocation, "", "location for the credits which will be used only if --retire-on-take falg is true")
```
Comment (ryanchristo) [inline x/ecocredit/client/basket/tx.go:74]:
```suggestion
	cmd.Flags().String(FlagRetirementLocation, "", "location for the credits which will be used only if --retire-on-take flag is true")
```

File: x/ecocredit/client/basket/tx.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+const (
+	FlagRetirementLocation = "retirement-location"
+	FlagRetireOnTake       = "retire-on-take"
+)
+
+func TxTakeFromBasket() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "take-from-basket [owner] [basket_denom] [amount]",
+		Short: "takes credits from a basket",
```
Comment (ryanchristo) [inline x/ecocredit/client/basket/tx.go:23]:
```suggestion
		Short: "Takes credits from a basket",
```

File: x/ecocredit/client/basket/tx.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+		Long: strings.TrimSpace(`takes credits from a basket starting from the oldest credits first.
+
+Parameters:
+		owner: account address of the owner of the basket.
+		Note, the'--from' flag is ignored as it is implied from [owner]
```
Comment (robert-zaremba) [inline x/ecocredit/client/basket/tx.go:28]: Why not removing `owner` param from this command, and use `--from` instead? Sounds to be less hacky.

File: x/ecocredit/client/basket/tx.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+		basket_denom: basket denom is the basket denom to take credits from.
```
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply x/ecocredit/client/basket/tx.go:29]: Or:
> asket_denom: basket identifier

**Review by robert-zaremba — DISMISSED** [review]
let's remove `owner` from CLI params and only use `--from`.

File: x/ecocredit/client/basket/tx.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+	cmd := &cobra.Command{
+		Use:   "take-from-basket [basket_denom] [amount]",
+		Short: "Takes credits from a basket",
+		Long: strings.TrimSpace(`takes credits from a basket starting from the oldest credits first.
+
+Parameters:
+		basket_denom: denom identifying basket from which we redeem credits.
+		amount: amount is a positive integer number of basket tokens to convert into credits.
+Flags:
+```suggestion
```
Comment (ryanchristo) [inline x/ecocredit/client/basket/tx.go:30]:
```suggestion
```

File: x/ecocredit/client/basket/tx.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
+package basket
```
Comment (ryanchristo) [inline x/ecocredit/client/basket/tx.go:1]:
```suggestion
package basketclient
```

**Pull request merged**